app/models/person.rb

```diff
@@ -1,6 +1,74 @@
 class Person < ActiveRecord::Base
-  has_one :location
+  has_one :location, autosave: true
   has_one :district, through: :location
+  has_one :representative, through: :district
+  has_one :target_rep, -> { targeted }, through: :district
+  has_one :state, through: :location
+  has_many :senators, through: :state
 
-  validates :email, presence: true, uniqueness: { case_sensitive: false }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, uniqueness: { case_sensitive: false },
+                    format: { with: VALID_EMAIL_REGEX }
+
+  before_save { self.email = email.downcase }
+
+  alias_method :location_association, :location
```

Comment:
I'm not sure I follow why there is the need to do the location saving work on Person instead of on location.  Seems like this creates a lot of work that could be offloaded onto ActiveRecord